action.yml

```diff
@@ -5,17 +5,52 @@ runs:
   using: composite
   steps:
 
+    - name: Install PHP
+      uses: shivammathur/setup-php@1a18b2267f80291a81ca1d33e7c851fe09e7dfc4 # v2.22.0
+      with:
+        php-version: 8.1
+
     - name: Generate tag name
       id: generate_tag_name
       shell: bash
       env:
         GITHUB_REF: ${{ github.ref }}
+        GITHUB_REPOSITORY: ${{ github.repository }}
       run: |
         # refs/tags/0.1.23 => 0.1.23
         TAG=$(echo $GITHUB_REF | cut -c 11-)
         if ! [[ $TAG =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
-          echo "Invalid semver tag $TAG"
-          exit 1
+          echo "Invalid semver tag $TAG, will use highest semver tag on repo instead"
+          # Gets 100 most recently created tags from GitHub API
+          # https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28
+          RESP_CODE=$(curl -w %{http_code} -s -o __tags.json \
+          -X GET "https://api.github.com/repos/$GITHUB_REPOSITORY/tags?per_page=100" \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer ${{ github.token }}" \
+          -H "X-GitHub-Api-Version: 2022-11-28")
+          if [[ $RESP_CODE != "200" ]]; then
+              echo "Unable to read list of tags - HTTP response code was $RESP_CODE"
+              cat __tags.json
+              exit 1
+          fi
+          # Get highest semver tag from list
+          TAG=$(php -r '
+              $json = json_decode(file_get_contents("__tags.json"), true);
+              $tags = [];
+              foreach ($json as $record) {
+                  if (preg_match("#^[0-9]+\.[0-9]+\.[0-9]+$#", $record["name"])) {
```

Comment:
This includes tags for the 0.x major release line - are we intentionally including that here? I vaguely recall we explicitly exclude 0.x in some scenarios so just want to check whether this is intentionally allowed here.

Reply:
I had considered that though we should allow 0.x tags as they are valid semver tags

There is logic to support 0.x tags - https://github.com/silverstripe/gha-auto-tag/blob/1/action.yml#L24